test/messaging/mapper_test.exs

```diff
@@ -112,4 +114,151 @@ defmodule Wanda.Messaging.MapperTest do
              } = Mapper.to_execution_completed(execution)
     end)
   end
+
+  test "should map from ExecutionRequestedV1 event" do
+    execution_id = UUID.uuid4()
+    group_id = UUID.uuid4()
+
+    execution =
+      ExecutionRequested.new!(
+        execution_id: execution_id,
+        group_id: group_id,
+        targets: [
+          %{
+            agent_id: "agent1",
+            checks: ["check_1", "check_2"]
+          },
+          %{
+            agent_id: "agent3",
+            checks: ["check_3", "check_4"]
+          }
+        ],
+        env: %{
+          some_string: %{
+            kind: {:string_value, "some_string"}
+          },
+          some_number: %{
+            kind: {:number_value, 10}
+          },
+          some_boolean: %{
+            kind: {:boolean_value, true}
+          }
+        }
+      )
+
+    assert %{
+             execution_id: ^execution_id,
+             group_id: ^group_id,
+             targets: [
+               %Executions.Target{
+                 agent_id: "agent1",
+                 checks: ["check_1", "check_2"]
+               },
+               %Executions.Target{
+                 agent_id: "agent3",
+                 checks: ["check_3", "check_4"]
+               }
+             ],
+             env: %{
+               some_string: "some_string",
+               some_number: 10,
+               some_boolean: true
+             }
+           } = Mapper.from_execution_requested(execution)
+  end
+
+  test "should map from FactsGathered event" do
+    execution_id = UUID.uuid4()
+    group_id = UUID.uuid4()
+    agent_id = UUID.uuid4()
+
+    facts =
+      FactsGathered.new!(
+        execution_id: execution_id,
+        group_id: group_id,
+        agent_id: agent_id,
+        facts_gathered: [
+          %{
+            check_id: "check1",
+            name: "string_value",
+            fact_value: {:value, %{kind: {:string_value, "some_string"}}}
+          },
+          %{
+            check_id: "check2",
+            name: "number_value",
+            fact_value: {:value, %{kind: {:number_value, 10}}}
+          },
+          %{
+            check_id: "check3",
+            name: "boolean_value",
+            fact_value: {:value, %{kind: {:boolean_value, true}}}
+          },
+          %{
+            check_id: "check4",
+            name: "list_value",
+            fact_value:
```

Comment:
I didn't create any factory here, as these are protobuf related types, and are not part of our domain. They only have sense in this particular tests

Reply:
super!

Reply:
Fair enough